ext/openssl/openssl.c

```diff
@@ -5213,6 +5375,33 @@ PHP_FUNCTION(openssl_get_cipher_methods)
 }
 /* }}} */
 
+/* {{{ proto array openssl_get_curve_names()
+   Return array of available elliptic curves */
+#ifdef HAVE_EVP_PKEY_EC
+PHP_FUNCTION(openssl_get_curve_names)
+{
+	EC_builtin_curve *curves = NULL;
+	const char *sname;
+	int i;
+
+	array_init(return_value);
```

Comment:
it should be done after EC_get_builtin_curves otherwise there will be memleak in case it failed (RETURN_FALSE...)